File: Server/src/MagazinesManager.ts
import { ITemplateItem } from "@spt/models/eft/common/tables/ITemplateItem"
import { AbstractModManager } from "./AbstractModManager"
import { Helper } from "./Helper"
import { Constants } from "./Constants"


export class MagazinesManager extends AbstractModManager
{
    protected configName: string = "MagazinesConfig"

    protected afterPostDB(): void
    {
        for(let itemKey in this.databaseTables.templates.items)
        {
            const item = this.databaseTables.templates.items[itemKey]

            if (item._props.Cartridges &&
                item._props.Cartridges.length > 0 &&
                item._props.ReloadMagType !== "InternalMagazine")
            {
                this.setMagLoadModifier(item)
            }
        }

        Helper.iterateConfigItems(this.databaseTables, this.config, this.setMag)
        
        console.log(`${Constants.ModTitle}: Magazines changes applied!`)
    }

    private setMagLoadModifier(mag: ITemplateItem)
    {
        const maxCount = mag._props.Cartridges[0]._max_count

        if (maxCount < 50)
        {
            mag._props.LoadUnloadModifier += this.config.magLoadPercent

            if (mag._props.LoadUnloadModifier < 0)
            {
                mag._props.LoadUnloadModifier *= this.config.magLoadMultiplier;
            }
        }
        else
        {
            mag._props.LoadUnloadModifier += this.config.largeMagLoadPercent

            if (mag._props.LoadUnloadModifier < 0)
            {
                mag._props.LoadUnloadModifier *= this.config.largeLoadMultiplier;
            }
        }

        if (mag._props.LoadUnloadModifier < -70)
        {
            mag._props.LoadUnloadModifier = -70;
        }
    }

    private setMag(item: ITemplateItem, config: any): void
    {
        item._props.LoadUnloadModifier = config.magLoadPercent
    }
}

File: Server/VAI-NRTP/Managers/MagazinesManager.cs
using System.Text.Json.Nodes;
using SPTarkov.DI.Annotations;
using SPTarkov.Server.Core.Models.Eft.Common.Tables;
using SPTarkov.Server.Core.Models.Spt.Tables;

namespace VAI.NRTP.Managers;

[Injectable(InjectionType.Singleton)]
public sealed class MagazinesManager : AbstractModManager
{
    protected override string ConfigName => "MagazinesConfig";

    protected override void AfterPostDb()
    {
        foreach (var item in DatabaseTables.Templates.Items.Values)
        {
            var props = item.Properties;
            if (props is null)
            {
                continue;
            }

            var cartridges = props.Cartridges;
            if (cartridges is null || !cartridges.Any())
            {
                continue;
            }

            if (string.Equals(props.ReloadMagType?.ToString(), "InternalMagazine", StringComparison.OrdinalIgnoreCase))
            {
                continue;
            }

            SetMagLoadModifier(props, cartridges);
        }

        Helper.IterateConfigItems(DatabaseTables, Config, SetMag);

        Constants.GetLogger().Info($"{Constants.ModTitle}: Magazines changes applied!");
    }

    private void SetMagLoadModifier(TemplateItemProperties props, IEnumerable<Slot> cartridges)
    {
        var first = cartridges.FirstOrDefault();
        var maxCount = first?.MaxCount;
        if (maxCount is null)
        {
            return;
        }

        var loadUnloadModifier = props.LoadUnloadModifier ?? 0;

        if (maxCount.Value < 50)
        {
            loadUnloadModifier += GetConfigNumber("magLoadPercent", 0);

            if (loadUnloadModifier < 0)
            {
                loadUnloadModifier *= GetConfigNumber("magLoadMultiplier", 1);
            }
        }
        else
        {
            loadUnloadModifier += GetConfigNumber("largeMagLoadPercent", 0);

            if (loadUnloadModifier < 0)
            {
                loadUnloadModifier *= GetConfigNumber("largeLoadMultiplier", 1);
            }
        }

        if (loadUnloadModifier < -50)
        {
            loadUnloadModifier = -50;
        }

        props.LoadUnloadModifier = loadUnloadModifier;
    }

    private void SetMag(TemplateItem item, JsonObject config, ModContext databaseTables, JsonNode? rootConfig)
    {
        var value = GetNumberValue(config["magLoadPercent"]);
        if (value is null)
        {
            return;
        }

        var props = item.Properties;
        if (props is null)
        {
            return;
        }

        props.LoadUnloadModifier = value.Value;
    }
}
